### PR TITLE
Adding autoencoder model for representation learning + Small ViT refactoring

### PR DESCRIPTION
**Pull request by alzaia**
This PR does the following:

- [X] Adds baseline autoencoder model for another way of doing representation learning prior to clustering algos and refactoring to adapt model class to AE (similarity measure between input and reconstructed output).
- [X] Small refactoring of ViT baseline for input type/shape.
- [X] Small notebook to test that the vit model methods work as expected when giving it batches from Office31 dataset

I decided to limit this PR to adding the AE class and the small refactoring of the ViT class model. Another PR will be opened for the AE exploration, training experiments and comparison with resnet and vit.

### CODE REVIEW

File: image_classification_simulation/models/autoencoder_baseline.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+        Returns
+        -------
+        torch.Tensor
+            Logit scores
+        """
+        # print(batch_images.shape)
+
+        z_x = self.encoder(batch_images)
+        print("shape of code: ", z_x.shape)
+        logits = self.decoder(z_x)
```
Comment (SinaSarparast) [inline image_classification_simulation/models/autoencoder_baseline.py:231]: what is the result of the decoder?

File: image_classification_simulation/models/autoencoder_baseline.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        batch_idx : int
+            index of the input batch
+
+        Returns
+        -------
+        typing.Any
+            returns loss and logit scores.
+        """
+        input_data, targets = batch
+        logits = self(input_data)  # calls the forward pass of the model
```
Comment (SinaSarparast) [inline image_classification_simulation/models/autoencoder_baseline.py:118]: I think "logits" has to be renamed to reconstructed_input.

Comment (alzaia) [reply image_classification_simulation/models/autoencoder_baseline.py]: Done on the latest commit

**Review by SinaSarparast — APPROVED** [review]
looks good
